Hustle-Hamster/main.cpp

```diff
@@ -36,86 +44,211 @@ int printHammy(){
     return 0; 
 }
 
-void TYPE(const string &p);
-int dialoge();
-void printReport();
-void delay(int time);
-void didActivity(string activity);
+/**
+ * Method to type out strings in terminal (animation)
+ * 
+ * @param p string to print
+*/
+void TYPE(const string &p){
+    cout << "       "; 
+    for (char c : p) {
+        cout << c << flush;
+        Sleep(30);
+    }
+    cout << endl;
 
+}
+
+/**
+ * Method to print out report (unformatted as of 14/8/24)
+ * 
+ * @param dailyEntry the daily Journal entry
+*/
+void printReport(Journal &dailyEntry){
+    stringstream dayRate;
+    stringstream sleepRate;
+    stringstream moodRate;
+
+    dayRate << "You rated your day a " << dailyEntry.getDayRating() << "/5";
+    TYPE(dayRate.str());
+    sleepRate << "You rated your sleep a " << dailyEntry.getSleepRating() << "/5";
+    TYPE(sleepRate.str());
+    moodRate << "You said your mood today was: " << dailyEntry.getMood();
+    TYPE(moodRate.str());
+
+    if(!dailyEntry.getActivities().empty()){
+        TYPE("Here are the activities you completed today:");
+        for (const string& activity : dailyEntry.getActivities()) {
+            TYPE(activity);
+        }
+    }else{
+        TYPE("You didn't complete any activities today!");
+        TYPE("Tomorrow is a new day and you can start fresh!");
+    }
 
-int main(){
-    TYPE("WELCOME TO HAMSTER HANGOUT");
-    delay(stdDelay);
-    printHammy();
-    TYPE("Hi there!! It's so good to see you <3");
-    TYPE("Lets see how your day went!");
-    cout << separater << "\n"; 
-    delay(stdDelay);
-    dialoge();
-    cout << separater << "\n";
-    printReport();
 
 }
 
-int dialoge(){
-    TYPE("We will start by doing some ratings! Everything is Rated out of 5, with 5 being the best!");
-    delay(stdDelay);
+/**
+ * Method to ask and set the users daily rating
+ * 
+ * @param dailyEntry the daily Journal entry
+*/
+void dailyRating(Journal &dailyEntry){
+    int temp; 
+    bool validResponse = false; 
     TYPE("How was your day? [1] Terrible [2] Bad [3] Fine [4] Good [5] AMAZING!");
-    cin >> dayRating;
-
-    printHammy();
-    didActivity("Exercise");
-    didActivity("Reading");
-    didActivity("Meditation");
-
+    while(!validResponse){
+        cin >> temp;
+        if(!cin.fail() && (temp>0 && temp<6)){
+            dailyEntry.setDayRating(temp);
+            validResponse = true;
+            break;
+        }else{
+            TYPE("Please enter a valid number between 1 and 5");
+            cin.clear();
+            cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
```

Comment:
`max()` function used here causes:
`Warning C4003 not enough arguments for function-like macro invocation 'max'`

Reply:
OK I think I have fixed it here, it just means that for the string there is a certain buffer size, and just need to be careful that user input won't exceed the buffer. Or find an alternative later down the line.